lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,456 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns
+short_description: Manage DNS configuration of an ESXI host system
+description:
+- This module can be used to configure DNS for the default TCP/IP stack on an ESXi host system.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+- This module is a replacement for the module C(vmware_dns_config)
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  type:
+    description:
+    - Type of IP assignment. Either C(dhcp) or C(static).
+    - A VMkernel adapter needs to be set to DHCP if C(type) is set to C(dhcp).
+    type: str
+    default: 'static'
+    choices: [ 'dhcp', 'static' ]
+    required: False
+  device:
+    description:
+    - The VMkernel network adapter to obtain DNS settings from.
+    - The parameter is only required in case of C(type) is set to C(dhcp).
+  host_name:
+    description:
+    - The hostname to be used for the ESXi host.
+    type: str
+    required: True
```

Comment:
These required parameters seem not required if i look at the module definition in the code? Only if the type is static? Probably best to update documentation accordingly for `host_name`, `domain`, `dns_servers` and `search_domains`. Will probably require some changes in the `ensure` method as well

Or if they are required, the module definition should be updated.